zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/JobControllerRoundRobinTest.java

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+
+import com.jayway.jsonpath.JsonPath;
+import io.camunda.service.JobServices;
+import io.camunda.zeebe.broker.client.api.BrokerClient;
+import io.camunda.zeebe.broker.client.api.dto.BrokerRejection;
+import io.camunda.zeebe.broker.client.api.dto.BrokerRejectionResponse;
+import io.camunda.zeebe.gateway.api.job.ActivateJobsStub;
+import io.camunda.zeebe.gateway.api.util.StubbedBrokerClient;
+import io.camunda.zeebe.gateway.impl.broker.request.BrokerActivateJobsRequest;
+import io.camunda.zeebe.gateway.impl.job.ActivateJobsHandler;
+import io.camunda.zeebe.gateway.impl.job.RoundRobinActivateJobsHandler;
+import io.camunda.zeebe.gateway.protocol.rest.JobActivationResponse;
+import io.camunda.zeebe.gateway.rest.ResponseMapper;
+import io.camunda.zeebe.gateway.rest.controller.util.ResettableJobActivationRequestResponseObserver;
+import io.camunda.zeebe.protocol.Protocol;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.scheduler.Actor;
+import io.camunda.zeebe.scheduler.ActorScheduler;
+import io.camunda.zeebe.scheduler.clock.ActorClock;
+import io.camunda.zeebe.scheduler.clock.ControlledActorClock;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.MediaType;
+import org.springframework.util.unit.DataSize;
+
+@WebMvcTest(JobController.class)
+public class JobControllerRoundRobinTest extends RestControllerTest {
```

Comment:
I saw this on your PR for completed jobs endpoint @tmetzke

Reply:
So this is a copy of the test created in the ["completeJob" endpoint PR](https://github.com/camunda/camunda/pull/20089)?

Reply:
If so, let's rather complete it over at the other PR, or is there a specific need to copy it over here?

Reply:
No, I copied from the other PR. I saw it and use it as a starting point. What's the status of the PR ? It's ready to be reviewed ? Git should be smart enough to not create conflicts on this test class, even if we merge this PR first

Reply:
I see. The other PR is not completely ready for review. I'll have to look at it again, I stopped close to being finished.
The test is however not required over here in this PR, is it? I don't mind in which PR it's contained but the classes are not entirely equal so git might have an issue with them. Let's either remove it here or in the other PR, I don't care much 😅

Reply:
I will remove the test class from this PR